app/controllers/reviews_controller.rb

```diff
@@ -48,10 +46,25 @@ def update
     end
   end
 
+  def vote
+    @review = Review.find(params[:id])
+    change = params[:change].to_i
+    if params[:poll] == "true"
+      @review.tally += 1 + change
+    elsif params[:poll] == "false"
+      @review.tally -= 1 + change
+    end
+   @review.save
+    redirect_to @review.shoe
+  end
```

Comment:
Nice controller method (though your use of non-RESTful routing is not necessarily a design decision with which I agree.)